tests/queries/0_stateless/01710_projection_with_nullable_keys.sql

```diff
@@ -0,0 +1,9 @@
+DROP TABLE IF EXISTS sales;
+
+CREATE TABLE sales (DATE_SOLD DateTime64(3, 'UTC'), PRODUCT_ID Nullable(String)) Engine MergeTree() PARTITION BY toYYYYMM(DATE_SOLD) ORDER BY DATE_SOLD;
+
+ALTER TABLE sales ADD PROJECTION test (SELECT toInt64(COUNT(*)) GROUP BY PRODUCT_ID, DATE_SOLD);
```

Comment:
I think this query should fail with `Sorting key contains nullable columns, but merge tree setting 'allow_nullable_key' is disabled`. This is expected when `allow_nullable_key` is not set.

But it should work after
```
alter table sales modify setting allow_nullable_key=1
```

Now it does not:
```
:) show create table sales

SHOW CREATE TABLE sales

Query id: 523f51e7-244c-4f8d-9445-b82e3718b43b

┌─statement───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CREATE TABLE default.sales
(
    `DATE_SOLD` DateTime64(3, 'UTC'),
    `PRODUCT_ID` Nullable(String)
)
ENGINE = MergeTree
PARTITION BY toYYYYMM(DATE_SOLD)
ORDER BY DATE_SOLD
SETTINGS index_granularity = 8192, allow_nullable_key = 1 │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

:) ALTER TABLE sales
    ADD PROJECTION test
    (
        SELECT toInt64(COUNT(*))
        GROUP BY
            PRODUCT_ID,
            DATE_SOLD
    )

ALTER TABLE sales
    ADD PROJECTION test
    (
        SELECT toInt64(COUNT(*))
        GROUP BY 
            PRODUCT_ID,
            DATE_SOLD
    )

Query id: 0101b5db-ccd3-4215-b70f-397d0b004d8d


0 rows in set. Elapsed: 0.002 sec. 

Received exception from server (version 23.9.1):
Code: 44. DB::Exception: Received from localhost:9000. DB::Exception: Sorting key contains nullable columns, but merge tree setting `allow_nullable_key` is disabled. (ILLEGAL_COLUMN)

```

And this ^ should be fixed.

Reply:
Btw, if you initially create a table with `allow_nullable_key = 1`, `ADD PROJECTION` works.

Reply:
Discussed it in TG.
The implementation looks ok if we look at projections as a kind of secondary index.
Let's make this change to make live easier.